bba/bba.go

```diff
@@ -0,0 +1,113 @@
+package bba
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/DE-labtory/cleisthenes/pb"
+
+	"github.com/DE-labtory/cleisthenes"
+)
+
+type Binary = bool
+
+const (
+	one  Binary = true
+	zero        = false
+)
+
+type request struct {
+	id   cleisthenes.ConnId
+	data *pb.BBA
+	err  chan error
+}
+
+type BBA struct {
+	*sync.RWMutex
+
+	// number of network nodes
+	n int
+	// number of byzantine nodes which can tolerate
+	f int
+
+	proposer cleisthenes.Member
+
+	stop int32
```

Comment:
Q) why this is int32?

Reply:
```
func (b *BBA) toDie() bool {
	return atomic.LoadInt32(&(b.stop)) == int32(1)
}
```
because of this function! `stop` is flag whether current bba instance is running or not